### PR TITLE
Update: tests/manage/test_ocs_324.py

### PR DESCRIPTION
**Pull request by sidhant-agrawal**
This PR addresses #262 and update the test tier.

Signed-off-by: Sidhant Agrawal <sagrawal@redhat.com>

### CODE REVIEW

**Comment by sidhant-agrawal** [comment]
```
========================================================== test session starts ==========================================================
platform linux -- Python 3.7.3, pytest-4.6.3, py-1.8.0, pluggy-0.12.0
rootdir: /home/sagrawal/Projects/redhat_github/ocs/ocs-ci-fork, inifile: pytest.ini
plugins: reportportal-1.0.5
collected 2 items                                                                                                                       

tests/manage/test_ocs_324.py::TestCaseOCS324::test_pvc_delete_create_same_name[CephBlockPool] 
------------------------------------------------------------ live log setup -------------------------------------------------------------
14:27:51 - MainThread - root - INFO - All logs are located under /tmp/logs
14:27:51 - MainThread - tests.conftest - INFO - Running OCS basic installation
14:27:51 - MainThread - oc.openshift_ops - INFO - Testing access to cluster with /home/sagrawal/ocs-runs/auth/kubeconfig
14:27:51 - MainThread - utility.utils - INFO - Executing command: oc cluster-info
14:27:54 - MainThread - oc.openshift_ops - INFO - Access to cluster is OK!
14:27:54 - MainThread - tests.conftest - INFO - The installation is skipped because the cluster is running
14:27:54 - MainThread - utility.utils - INFO - Executing command: oc --kubeconfig /home/sagrawal/ocs-runs/auth/kubeconfig get Pod -A -o yaml
14:28:16 - MainThread - utility.utils - INFO - Executing command: oc --kubeconfig /home/sagrawal/ocs-runs/auth/kubeconfig get StorageClass -A -o yaml
14:28:19 - MainThread - utility.utils - INFO - Executing command: oc --kubeconfig /home/sagrawal/ocs-runs/auth/kubeconfig get CephFileSystem -A -o yaml
14:28:21 - MainThread - utility.utils - INFO - Executing command: oc --kubeconfig /home/sagrawal/ocs-runs/auth/kubeconfig get CephBlockPool -A -o yaml
14:28:23 - MainThread - utility.utils - INFO - Executing command: oc --kubeconfig /home/sagrawal/ocs-runs/auth/kubeconfig get PersistentVolume -A -o yaml
14:28:24 - MainThread - utility.utils - INFO - Executing command: oc --kubeconfig /home/sagrawal/ocs-runs/auth/kubeconfig get PersistentVolumeClaim -A -o yaml
14:28:26 - MainThread - utility.utils - INFO - Executing command: oc --kubeconfig /home/sagrawal/ocs-runs/auth/kubeconfig get Secret -A -o yaml
14:28:59 - MainThread - utility.utils - INFO - Executing command: oc --kubeconfig /home/sagrawal/ocs-runs/auth/kubeconfig get Namespace -A -o yaml
14:29:01 - MainThread - tests.manage.test_ocs_324 - INFO - Creating resources for CephBlockPool interface
14:29:01 - MainThread - resources.ocs - INFO - Adding CephBlockPool with name cbp-test-2314290197
14:29:01 - MainThread - utility.utils - INFO - Executing command: oc -n openshift-storage --kubeconfig /home/sagrawal/ocs-runs/auth/kubeconfig create -f /tmp/CephBlockPoolv6flo9zp -o yaml
14:29:04 - MainThread - utility.utils - INFO - Executing command: oc -n openshift-storage --kubeconfig /home/sagrawal/ocs-runs/auth/kubeconfig get CephBlockPool cbp-test-2314290197 -o yaml
14:29:06 - MainThread - tests.helpers - INFO - Verifying that block pool cbp-test-2314290197 exists
14:29:06 - MainThread - utility.utils - INFO - Executing command: oc -n openshift-storage --kubeconfig /home/sagrawal/ocs-runs/auth/kubeconfig get Pod --selector=app=rook-ceph-tools -o yaml
14:29:11 - MainThread - utility.utils - INFO - Executing command: oc -n openshift-storage --kubeconfig /home/sagrawal/ocs-runs/auth/kubeconfig rsh rook-ceph-tools-76c7d559b6-99rbx ceph osd lspools --format json-pretty
14:29:17 - MainThread - utility.utils - INFO - Executing command: oc -n openshift-storage --kubeconfig /home/sagrawal/ocs-runs/auth/kubeconfig get Pod --selector=app=rook-ceph-tools -o yaml
14:29:18 - MainThread - utility.utils - INFO - Executing command: oc -n openshift-storage --kubeconfig /home/sagrawal/ocs-runs/auth/kubeconfig rsh rook-ceph-tools-76c7d559b6-99rbx ceph auth get-key client.admin --format json-pretty
14:29:30 - MainThread - resources.ocs - INFO - Adding Secret with name secret-test-2314293006
14:29:30 - MainThread - utility.utils - INFO - Executing command: oc -n openshift-storage --kubeconfig /home/sagrawal/ocs-runs/auth/kubeconfig create -f /tmp/Secret5xj14749 -o yaml
14:29:32 - MainThread - utility.utils - INFO - Executing command: oc -n openshift-storage --kubeconfig /home/sagrawal/ocs-runs/auth/kubeconfig get Secret secret-test-2314293006 -o yaml
14:29:34 - MainThread - resources.ocs - INFO - Adding StorageClass with name storageclass-test-2314293407
14:29:34 - MainThread - utility.utils - INFO - Executing command: oc -n openshift-storage --kubeconfig /home/sagrawal/ocs-runs/auth/kubeconfig create -f /tmp/StorageClasst3i7n2x9 -o yaml
14:29:36 - MainThread - utility.utils - INFO - Executing command: oc -n openshift-storage --kubeconfig /home/sagrawal/ocs-runs/auth/kubeconfig get StorageClass storageclass-test-2314293407 -o yaml
------------------------------------------------------------- live log call -------------------------------------------------------------
14:29:37 - MainThread - resources.ocs - INFO - Adding PersistentVolumeClaim with name pvc-test-2314293764
14:29:37 - MainThread - utility.utils - INFO - Executing command: oc -n openshift-storage --kubeconfig /home/sagrawal/ocs-runs/auth/kubeconfig create -f /tmp/PersistentVolumeClaimva6b43wg -o yaml
14:29:39 - MainThread - utility.utils - INFO - Executing command: oc -n openshift-storage --kubeconfig /home/sagrawal/ocs-runs/auth/kubeconfig get PersistentVolumeClaim pvc-test-2314293764 -o yaml
14:29:41 - MainThread - utility.utils - INFO - Executing command: oc -n openshift-storage --kubeconfig /home/sagrawal/ocs-runs/auth/kubeconfig get PersistentVolumeClaim pvc-test-2314293764 -o yaml
14:29:42 - MainThread - utility.utils - INFO - Executing command: oc -n openshift-storage --kubeconfig /home/sagrawal/ocs-runs/auth/kubeconfig get PersistentVolumeClaim pvc-test-2314293764 -o yaml
14:29:44 - MainThread - utility.utils - INFO - Executing command: oc -n openshift-storage --kubeconfig /home/sagrawal/ocs-runs/auth/kubeconfig get PersistentVolume pvc-3b5d8939-9595-11e9-8418-02697af25b98 -o yaml
14:29:46 - MainThread - tests.manage.test_ocs_324 - INFO - Deleting PersistentVolumeClaim pvc-test-2314293764
14:29:46 - MainThread - utility.utils - INFO - Executing command: oc -n openshift-storage --kubeconfig /home/sagrawal/ocs-runs/auth/kubeconfig delete PersistentVolumeClaim pvc-test-2314293764
14:29:48 - MainThread - tests.manage.test_ocs_324 - INFO - Creating PersistentVolumeClaim with same name pvc-test-2314293764
14:29:48 - MainThread - resources.ocs - INFO - Adding PersistentVolumeClaim with name pvc-test-2314293764
14:29:48 - MainThread - utility.utils - INFO - Executing command: oc -n openshift-storage --kubeconfig /home/sagrawal/ocs-runs/auth/kubeconfig create -f /tmp/PersistentVolumeClaimc3v7r_c7 -o yaml
14:29:50 - MainThread - utility.utils - INFO - Executing command: oc -n openshift-storage --kubeconfig /home/sagrawal/ocs-runs/auth/kubeconfig get PersistentVolumeClaim pvc-test-2314293764 -o yaml
14:29:51 - MainThread - utility.utils - INFO - Executing command: oc -n openshift-storage --kubeconfig /home/sagrawal/ocs-runs/auth/kubeconfig get PersistentVolumeClaim pvc-test-2314293764 -o yaml
14:29:53 - MainThread - utility.utils - INFO - Executing command: oc -n openshift-storage --kubeconfig /home/sagrawal/ocs-runs/auth/kubeconfig get PersistentVolumeClaim pvc-test-2314293764 -o yaml
14:29:55 - MainThread - utility.utils - INFO - Executing command: oc -n openshift-storage --kubeconfig /home/sagrawal/ocs-runs/auth/kubeconfig get PersistentVolume pvc-41b7ebe0-9595-11e9-8418-02697af25b98 -o yaml
PASSED                                                                                                                            [ 50%]
----------------------------------------------------------- live log teardown -----------------------------------------------------------
14:29:57 - MainThread - tests.manage.test_ocs_324 - INFO - Deleting resources for CephBlockPool interface
14:29:57 - MainThread - tests.manage.test_ocs_324 - INFO - Deleting PersistentVolumeClaim pvc-test-2314293764
14:29:57 - MainThread - utility.utils - INFO - Executing command: oc -n openshift-storage --kubeconfig /home/sagrawal/ocs-runs/auth/kubeconfig delete PersistentVolumeClaim pvc-test-2314293764
14:30:00 - MainThread - tests.manage.test_ocs_324 - INFO - Deleting StorageClass storageclass-test-2314293407
14:30:00 - MainThread - utility.utils - INFO - Executing command: oc --kubeconfig /home/sagrawal/ocs-runs/auth/kubeconfig delete StorageClass storageclass-test-2314293407
14:30:02 - MainThread - tests.manage.test_ocs_324 - INFO - Deleting Secret secret-test-2314293006
14:30:02 - MainThread - utility.utils - INFO - Executing command: oc -n openshift-storage --kubeconfig /home/sagrawal/ocs-runs/auth/kubeconfig delete Secret secret-test-2314293006
14:30:05 - MainThread - tests.manage.test_ocs_324 - INFO - Deleting CephBlockPool cbp-test-2314290197
14:30:05 - MainThread - utility.utils - INFO - Executing command: oc -n openshift-storage --kubeconfig /home/sagrawal/ocs-runs/auth/kubeconfig delete CephBlockPool cbp-test-2314290197

tests/manage/test_ocs_324.py::TestCaseOCS324::test_pvc_delete_create_same_name[CephFileSystem] 
------------------------------------------------------------ live log setup -------------------------------------------------------------
14:30:20 - MainThread - tests.manage.test_ocs_324 - INFO - Creating resources for CephFileSystem interface
14:30:20 - MainThread - utility.utils - INFO - Executing command: oc -n openshift-storage --kubeconfig /home/sagrawal/ocs-runs/auth/kubeconfig get Pod --selector=app=rook-ceph-tools -o yaml
14:30:23 - MainThread - utility.utils - INFO - Executing command: oc -n openshift-storage --kubeconfig /home/sagrawal/ocs-runs/auth/kubeconfig rsh rook-ceph-tools-76c7d559b6-99rbx ceph fs ls --format json-pretty
14:30:32 - MainThread - utility.utils - INFO - Executing command: oc -n openshift-storage --kubeconfig /home/sagrawal/ocs-runs/auth/kubeconfig get Pod --selector=app=rook-ceph-tools -o yaml
14:30:33 - MainThread - utility.utils - INFO - Executing command: oc -n openshift-storage --kubeconfig /home/sagrawal/ocs-runs/auth/kubeconfig rsh rook-ceph-tools-76c7d559b6-99rbx ceph auth get-key client.admin --format json-pretty
14:30:39 - MainThread - resources.ocs - INFO - Adding Secret with name secret-test-2314303992
14:30:39 - MainThread - utility.utils - INFO - Executing command: oc -n openshift-storage --kubeconfig /home/sagrawal/ocs-runs/auth/kubeconfig create -f /tmp/Secret4f0skuyp -o yaml
14:30:41 - MainThread - utility.utils - INFO - Executing command: oc -n openshift-storage --kubeconfig /home/sagrawal/ocs-runs/auth/kubeconfig get Secret secret-test-2314303992 -o yaml
14:30:44 - MainThread - resources.ocs - INFO - Adding StorageClass with name storageclass-test-2314304408
14:30:44 - MainThread - utility.utils - INFO - Executing command: oc -n openshift-storage --kubeconfig /home/sagrawal/ocs-runs/auth/kubeconfig create -f /tmp/StorageClass_5qf5zqq -o yaml
14:30:45 - MainThread - utility.utils - INFO - Executing command: oc -n openshift-storage --kubeconfig /home/sagrawal/ocs-runs/auth/kubeconfig get StorageClass storageclass-test-2314304408 -o yaml
------------------------------------------------------------- live log call -------------------------------------------------------------
14:30:47 - MainThread - resources.ocs - INFO - Adding PersistentVolumeClaim with name pvc-test-2314304733
14:30:47 - MainThread - utility.utils - INFO - Executing command: oc -n openshift-storage --kubeconfig /home/sagrawal/ocs-runs/auth/kubeconfig create -f /tmp/PersistentVolumeClaimt4hz_am_ -o yaml
14:30:49 - MainThread - utility.utils - INFO - Executing command: oc -n openshift-storage --kubeconfig /home/sagrawal/ocs-runs/auth/kubeconfig get PersistentVolumeClaim pvc-test-2314304733 -o yaml
14:30:50 - MainThread - utility.utils - INFO - Executing command: oc -n openshift-storage --kubeconfig /home/sagrawal/ocs-runs/auth/kubeconfig get PersistentVolumeClaim pvc-test-2314304733 -o yaml
14:30:53 - MainThread - utility.utils - INFO - Executing command: oc -n openshift-storage --kubeconfig /home/sagrawal/ocs-runs/auth/kubeconfig get PersistentVolumeClaim pvc-test-2314304733 -o yaml
14:30:56 - MainThread - utility.utils - INFO - Executing command: oc -n openshift-storage --kubeconfig /home/sagrawal/ocs-runs/auth/kubeconfig get PersistentVolume pvc-6507afd7-9595-11e9-9079-0a4c2a7c4d50 -o yaml
14:30:59 - MainThread - tests.manage.test_ocs_324 - INFO - Deleting PersistentVolumeClaim pvc-test-2314304733
14:30:59 - MainThread - utility.utils - INFO - Executing command: oc -n openshift-storage --kubeconfig /home/sagrawal/ocs-runs/auth/kubeconfig delete PersistentVolumeClaim pvc-test-2314304733
14:31:02 - MainThread - tests.manage.test_ocs_324 - INFO - Creating PersistentVolumeClaim with same name pvc-test-2314304733
14:31:02 - MainThread - resources.ocs - INFO - Adding PersistentVolumeClaim with name pvc-test-2314304733
14:31:02 - MainThread - utility.utils - INFO - Executing command: oc -n openshift-storage --kubeconfig /home/sagrawal/ocs-runs/auth/kubeconfig create -f /tmp/PersistentVolumeClaimtg2mi2ns -o yaml
14:31:04 - MainThread - utility.utils - INFO - Executing command: oc -n openshift-storage --kubeconfig /home/sagrawal/ocs-runs/auth/kubeconfig get PersistentVolumeClaim pvc-test-2314304733 -o yaml
14:31:05 - MainThread - utility.utils - INFO - Executing command: oc -n openshift-storage --kubeconfig /home/sagrawal/ocs-runs/auth/kubeconfig get PersistentVolumeClaim pvc-test-2314304733 -o yaml
14:31:06 - MainThread - utility.utils - INFO - Executing command: oc -n openshift-storage --kubeconfig /home/sagrawal/ocs-runs/auth/kubeconfig get PersistentVolumeClaim pvc-test-2314304733 -o yaml
14:31:08 - MainThread - utility.utils - INFO - Executing command: oc -n openshift-storage --kubeconfig /home/sagrawal/ocs-runs/auth/kubeconfig get PersistentVolume pvc-6ddf0791-9595-11e9-9079-0a4c2a7c4d50 -o yaml
PASSED                                                                                                                            [100%]
----------------------------------------------------------- live log teardown -----------------------------------------------------------
14:31:13 - MainThread - tests.manage.test_ocs_324 - INFO - Deleting resources for CephFileSystem interface
14:31:13 - MainThread - tests.manage.test_ocs_324 - INFO - Deleting PersistentVolumeClaim pvc-test-2314304733
14:31:13 - MainThread - utility.utils - INFO - Executing command: oc -n openshift-storage --kubeconfig /home/sagrawal/ocs-runs/auth/kubeconfig delete PersistentVolumeClaim pvc-test-2314304733
14:31:18 - MainThread - tests.manage.test_ocs_324 - INFO - Deleting StorageClass storageclass-test-2314304408
14:31:18 - MainThread - utility.utils - INFO - Executing command: oc --kubeconfig /home/sagrawal/ocs-runs/auth/kubeconfig delete StorageClass storageclass-test-2314304408
14:31:21 - MainThread - tests.manage.test_ocs_324 - INFO - Deleting Secret secret-test-2314303992
14:31:21 - MainThread - utility.utils - INFO - Executing command: oc -n openshift-storage --kubeconfig /home/sagrawal/ocs-runs/auth/kubeconfig delete Secret secret-test-2314303992
14:31:23 - MainThread - utility.utils - INFO - Executing command: oc --kubeconfig /home/sagrawal/ocs-runs/auth/kubeconfig get Pod -A -o yaml
14:31:34 - MainThread - utility.utils - INFO - Executing command: oc --kubeconfig /home/sagrawal/ocs-runs/auth/kubeconfig get StorageClass -A -o yaml
14:31:35 - MainThread - utility.utils - INFO - Executing command: oc --kubeconfig /home/sagrawal/ocs-runs/auth/kubeconfig get CephFileSystem -A -o yaml
14:31:37 - MainThread - utility.utils - INFO - Executing command: oc --kubeconfig /home/sagrawal/ocs-runs/auth/kubeconfig get CephBlockPool -A -o yaml
14:31:39 - MainThread - utility.utils - INFO - Executing command: oc --kubeconfig /home/sagrawal/ocs-runs/auth/kubeconfig get PersistentVolume -A -o yaml
14:31:41 - MainThread - utility.utils - INFO - Executing command: oc --kubeconfig /home/sagrawal/ocs-runs/auth/kubeconfig get PersistentVolumeClaim -A -o yaml
14:31:43 - MainThread - utility.utils - INFO - Executing command: oc --kubeconfig /home/sagrawal/ocs-runs/auth/kubeconfig get Secret -A -o yaml
14:32:34 - MainThread - utility.utils - INFO - Executing command: oc --kubeconfig /home/sagrawal/ocs-runs/auth/kubeconfig get Namespace -A -o yaml


====================================================== 2 passed in 288.68 seconds =======================================================
```

File: tests/manage/test_ocs_324.py [hunk 1]
```diff
@@ -92,11 +94,24 @@ def test_pvc_delete_create_same_name(self, test_fixture):
...
+        logger.info(f"Deleting {PVC_OBJ.kind} {PVC_OBJ.name}")
         assert PVC_OBJ.delete(), f"Failed to delete PVC"
+
+        logger.info(f"Creating {PVC_OBJ.kind} with same name {PVC_OBJ.name}")
         PVC_OBJ = helpers.create_pvc(
             sc_name=self.sc_obj.name, pvc_name=PVC_OBJ.name
         )
-        logger.info(
-            f"PersistentVolumeClaim created with same name {PVC_OBJ.name}"
+        backed_pv = PVC_OBJ.get().get('spec').get('volumeName')
```
Comment (RazTamir) [inline tests/manage/test_ocs_324.py]: backed_pv = PVC_OBJ.backed_pv

File: tests/manage/test_ocs_324.py [hunk 2]
```diff
@@ -92,11 +94,24 @@ def test_pvc_delete_create_same_name(self, test_fixture):
         global PVC_OBJ
 
         PVC_OBJ = helpers.create_pvc(sc_name=self.sc_obj.name)
-        logger.info(f"Deleting PersistentVolumeClaim with name {PVC_OBJ.name}")
+        pv_obj = ocp.OCP(
+            kind=constants.PV, namespace=config.ENV_DATA['cluster_namespace']
+        )
+        backed_pv = PVC_OBJ.get().get('spec').get('volumeName')
```
Comment (RazTamir) [inline tests/manage/test_ocs_324.py]: backed_pv = PVC_OBJ.backed_pv

File: tests/manage/test_ocs_324.py (hunk 1 above)
Comment (sidhant-agrawal) [reply tests/manage/test_ocs_324.py]: @RazTamir I'm using helpers.py to create PVC. The PVC_OBJ here does not have 'backed_pv' attribute.
Am I missing something?

Comment (RazTamir) [reply tests/manage/test_ocs_324.py]: ok.

+1 on the patch - the issue is with the helper function

File: tests/manage/test_ocs_324.py (hunk 2 above)
Comment (sidhant-agrawal) [reply tests/manage/test_ocs_324.py]: Resolving based on the above conversation

File: tests/manage/test_ocs_324.py [hunk 3]
```diff
@@ -32,26 +33,23 @@ def setup(self):
...
         self.interface_name = self.cbp_obj.name
 
     elif self.interface_type == constants.CEPHFILESYSTEM:
-        assert helpers.create_cephfilesystem(), (
-            f"Failed to create Ceph File System"
-        )
         self.interface_name = helpers.get_cephfs_data_pool_name()
 
+    self.secret_obj = helpers.create_secret(interface_type=self.interface_type)
+    assert self.secret_obj, f"Failed to create secret"
```
Comment (fbalak) [inline tests/manage/test_ocs_324.py]:
```suggestion
    assert self.secret_obj, 'Failed to create secret'
```

**Pull request merged**